tests/test_bot.py

```diff
@@ -0,0 +1,139 @@
+from flexmock import flexmock
+
+from release_bot.configuration import configuration
+from release_bot.releasebot import ReleaseBot
+from github_utils import GithubUtils, RELEASE_CONF
+
+
+@pytest.mark.skipif(not GithubUtils.github_api_status(), reason="Github api is down")
+@pytest.mark.skipif(not os.environ.get('GITHUB_TOKEN') and not os.environ.get('GITHUB_USER'),
+                    reason="missing GITHUB_TOKEN and GITHUB_USER variables")
+class TestBot:
+    """ Tests parts of bot workflow"""
+    github_token = os.environ.get('GITHUB_TOKEN')
+    github_user = os.environ.get('GITHUB_USER')
+    headers = {'Authorization': f'token {github_token}'}
+
+    def setup_method(self):
+        """ setup any state tied to the execution of the given method in a
+        class.  setup_method is invoked for every test method of a class.
+        """
+        configuration.set_logging(level=10)
+        configuration.debug = True
+
+        self.g_utils = GithubUtils(self.github_token, self.github_user)
+
+        self.g_utils.create_repo()
+        self.g_utils.setup_repo()
+
+        # set conf
+        configuration.repository_name = self.g_utils.repo
+        configuration.repository_owner = self.github_user
+        configuration.github_token = self.github_token
+        configuration.github_username = self.github_user
+        configuration.refresh_interval = 1
+
+        self.release_bot = ReleaseBot(configuration)
+
+    def teardown_method(self):
+        """ teardown any state that was previously setup with a setup_method
+        call.
+        """
+        if self.g_utils.repo:
+            self.g_utils.delete_repo()
+
+    @pytest.fixture()
+    def open_issue(self):
+        """Opens release issue in a repository"""
+        return self.g_utils.open_issue("0.0.1 release")
+
+    @pytest.fixture()
+    def multiple_release_issues(self):
+        """Opens two release issues in a repository"""
+        self.g_utils.open_issue("0.0.1 release")
+        self.g_utils.open_issue("0.0.2 release")
+
+    @pytest.fixture()
+    def open_pr(self):
+        """Opens two release issues in a repository"""
+        conf = yaml.safe_load(RELEASE_CONF) or {}
+        self.release_bot.new_release.update(conf)
+        self.open_issue()
+        self.release_bot.find_open_release_issues()
+        self.release_bot.make_release_pull_request()
+        pr_number = self.release_bot.github.pr_exists("0.0.1 release")
+        assert pr_number
+        assert self.g_utils.merge_pull_request(pr_number)
+
+    @pytest.fixture()
+    def github_release(self):
+        """Setups environment for releasing on Github"""
+        self.open_pr()
+        assert self.release_bot.find_newest_release_pull_request()
+        self.release_bot.make_new_github_release()
+        assert self.release_bot.github.latest_release() == "0.0.1"
+
+    @pytest.fixture()
+    def mock_upload(self):
+        """Mocks upload to PyPi"""
+        flexmock(self.release_bot.pypi, upload=True)
```

Comment:
Am I correct that this is a [shorthand](https://flexmock.readthedocs.io/en/latest/user-guide/#shorthand) to
```python
flexmock(self.release_bot.pypi).should_receive('upload').and_return('True')
```
?

Reply:
Yes it is.